packages/page-explorer/src/BlockInfo/ByHash.tsx

```diff
@@ -102,12 +103,53 @@ function BlockByHash ({ className = '', error, value }: Props): React.ReactEleme
       });
   }, [api, mountedRef, value]);
 
+  useEffect((): (() => void) | undefined => {
+    if (!mountedRef.current || !getHeader?.number) {
+      return;
+    }
+
+    const nextBlockNumber = getHeader.number.unwrap().addn(1);
+    let unsub: (() => void) | undefined;
+
+    api.rpc.chain.getBlockHash(nextBlockNumber)
+      .then((hash) => {
+        if (!hash.isEmpty) {
+          setState((prev) => ({
+            ...prev,
+            nextBlockHash: hash
+          }));
+        } else {
+          api.derive.chain.subscribeNewHeads((header: HeaderExtended): void => {
```

Comment:
It would be helpful to add a brief comment explaining this piece of logic.

```suggestion
          // Subscribes to the node until it no longer has a next block, and then unsubscribes.
          api.derive.chain.subscribeNewHeads((header: HeaderExtended): void => {
```

Reply:
Added in [1e77515](https://github.com/polkadot-js/apps/pull/11614/commits/1e77515d511bfaf06c4b80abd75479854d19e4e4)